docs/apache-airflow/administration-and-deployment/lineage.rst

```diff
@@ -89,6 +89,42 @@ has outlets defined (e.g. by using ``add_outlets(..)`` or has out of the box sup
 
 .. _precedence: https://docs.python.org/3/reference/expressions.html
 
+Hook Lineage
+------------
+
+Airflow provides a powerful feature for tracking data lineage not only between tasks but also from hooks used within those tasks.
+This functionality helps you understand how data flows throughout your Airflow pipelines.
+
+A global instance of ``HookLineageCollector`` serves as the central hub for collecting lineage information.
+Hooks can send details about datasets they interact with to this collector.
+The collector then uses this data to construct AIP-60 compliant Datasets, a standard format for describing datasets.
+
+.. code-block:: python
+
+    from airflow.lineage.hook_lineage import get_hook_lineage_collector
+
+
+    class CustomHook(BaseHook):
+        def run(self):
+            # run actual code
+            collector = get_hook_lineage_collector()
+            collector.add_inlet(dataset_kwargs={"scheme": "file", "path": "/tmp/in"}, self)
+            collector.add_outlet(dataset_kwargs={"scheme": "file", "path": "/tmp/out"}, self)
+
+Lineage data collected by the ``HookLineageCollector`` can be accessed using an instance of ``HookLineageReader``.
+
+.. code-block:: python
+
+    from airflow.lineage.hook_lineage import HookLineageReader
+
+
+    class CustomHookLineageReader(HookLineageReader):
+        def get_inputs(self):
+            return self.lineage_collector.collected_datasets.inputs
+
+If no ``HookLineageReader`` is registered within Airflow, a default ``NoOpCollector`` is used instead.
```

Comment:
We are missing details on how to register them.

On that note, feels a little weird that they have to come from providers? I haven't though through it much, but that's my initial reaction.

Reply:
I changed hook lineage readers to be a part of plugins now. Docs changed accordingly. Please let me know if that's more suitable place for it.